app/main/posts/modify/post-media.directive.js

```diff
@@ -17,84 +19,44 @@ function (
         require: '^^form',
         scope: {
             mediaId: '=',
+            media: '=',
             name: '@'
         },
         templateUrl: 'templates/main/posts/modify/media.html',
         link: function ($scope, element, attr, formCtrl) {
-            // Initialize file container
-            $scope.fileContainer = {file: null};
+            $scope.showAdd = function () {
+                return (!$scope.media.id && !$scope.media.changed);
+            };
 
-            // Initialize media object
-            $scope.media = {};
+            $scope.showReplace = function () {
+                return $scope.media.dataURI || $scope.media.id;
+            };
+
+            $scope.showDelete = function () {
+                return $scope.media.id;
+            };
 
             if ($scope.mediaId) {
                 MediaEndpoint.get({id: $scope.mediaId}).$promise.then(function (media) {
                     $scope.media = media;
```

Comment:
Maybe I'm wrong.. I've never 100% figure out how angular handles this.. and going back to things I've read about angular scope and inheritance I can't find support material

My thinking was that angular passed a JS object reference across the directive boundary.. and updates to the JS object would propagate back to the parent because it was literally the same object, where actually doing `scope.media = ...` would overwrite the reference entirely. **but** I think I'm just wrong. Googling suggests isolate scope's two way binding is implemented with a watch..

TLDR: ignore me, please continue :)